src/main/resources/templates/main.html

```diff
@@ -1,47 +1,96 @@
+			document.getElementById(embedChartButtonId).textContent = (embedChart.style.display === "none" ? "Embed chart" : "Hide embed code");
+		}
+		function changeChart(chartFrameId, chartSelectorId) {
+			document.getElementById('chartIFrame').setAttribute('src', document.getElementById('chartSelector').value);
+			document.getElementById('embedChart').textContent =
+				'<iframe width="600" height="400" seamless frameBorder="0" scrolling="no" src="' + document.getElementById('chartSelector').value + '&height=400"></iframe>';
+		}
+	</script>
 </head>
 <body>
 	<h1 th:text="#{main.title}"></h1>
 	<p>
 		<a th:href="${data.fileUrl}" th:text="${data.fileName}"></a>
 		<span th:text="${'('+ data.fileSize + ')'}"></span>
 	</p>
-	<table th:if="${!data.columns.isEmpty()}">
-		<thead>
-			<tr>
-				<th>Column</th>
-				<th>Type</th>
-			</tr>
-		</thead>
-		<tbody>
-			<tr th:each="column: ${data.columns}">
-				<td th:text="${column.name}"></td>
-				<td th:text="${column.type}"></td>
-			</tr>
-		</tbody>
-	</table>
 
 	<th:block th:switch="${data.status.name()}">
 		<div th:case="'READY'">
+			<table th:if="${!data.columns.isEmpty()}">
+				<thead>
+				<tr>
+					<th>Column</th>
+					<th>Type</th>
+				</tr>
+				</thead>
+				<tbody>
+				<tr th:each="column: ${data.columns}">
+					<td th:text="${column.name}"></td>
+					<td th:text="${column.type}"></td>
+				</tr>
+				</tbody>
+			</table>
 			<form method="POST" action="#" th:action="@{/dataverse-superset}">
 				<input type="hidden" id="name" name="datasetName" th:value="${data.getName()}">
 				<button type="submit">Import to Superset</button>
 			</form>
 		</div>
+
 		<div th:case="'IN_PROGRESS'">
 			<p>Import to Superset in progress... <button onClick="window.location.reload();">Refresh</button></p>
 		</div>
+
 		<div th:case="'COMPLETE'">
 			<p>
-				Import to Superset complete! ID: <a
+				Dataset imported to superset! Explore the data:  <a
 					th:href="${@environment.getProperty('superset.uri') + 'superset/explore/table/' + data.datasetId}"
 					th:text="${@environment.getProperty('superset.uri') + 'superset/explore/table/' + data.datasetId}"></a>
 			</p>
+
+			<div th:if="${chartToUrl != null && chartToUrl.size() > 0}">
+				Related charts:
+				<select id="chartSelector" th:field="*{chartToUrl}" onchange="changeChart('chartIFrame', 'chartSelector');">
+					<option th:each="entry : ${chartToUrl.entrySet()}" th:value="${@environment.getProperty('superset.uri') + entry.value}" th:text="${entry.key}"></option>
+				</select>
+				<button id="showHideEmbedChart" onclick="showHideEmbedCode('embedChart', 'showHideEmbedChart');">Embed chart</button>
+				<p id="embedChart" class="embedCode" style="display:none;">
+					&lt;iframe width="600" height="400" seamless frameBorder="0" scrolling="no" src="http://127.0.0.1:8088//superset/explore/?form_data=%7B%22slice_id%22%3A%20839%7D&amp;standalone=1&amp;height=400"&gt;&lt;/iframe&gt;
```

Comment:
Why hardcoded snippet? It's actually visible when the page loads and the first chart is selected by default!

Reply:
The embedChart paragraph code is empty then. I putted changeChart method into the onload attribute. Iframe is also simplified that way.